hammr/utils/publish_builders.py

```diff
@@ -0,0 +1,448 @@
+        return
+    if not "imageName" in builder:
+        printer.out("imageName in vcd builder not found", printer.ERROR)
+        return
+
+    pimage.credAccount.organizationName = builder["orgName"]
+    pimage.credAccount.catalogId = builder["catalogName"]
+    pimage.credAccount.displayName = builder["imageName"]
+    return pimage
+
+
+def publish_vcenter(builder):
+    pimage = PublishImageVSphere()
+
+    # doing field verification
+
+    if not "displayName" in builder:
+        printer.out("displayName in vcenter builder not found", printer.ERROR)
+        return
+
+    if not "esxHost" in builder:
+        printer.out("esxHost in vcenter builder not found", printer.ERROR)
+        return
+
+    if not "datastore" in builder:
+        printer.out("datastore in vcenter builder not found", printer.ERROR)
+        return
+
+    if "network" in builder:
+        pimage.network = builder["network"]
+
+    pimage.displayName = builder["displayName"]
+    pimage.esxHost = builder["esxHost"]
+    pimage.datastore = builder["datastore"]
+    return pimage
+
+
+def publish_cloudstack(builder):
+    pimage = PublishImageCloudStack()
+
+    # doing field verification
+    if not "imageName" in builder:
+        printer.out("imageName in cloudstack builder not found", printer.ERROR)
+        return
+    if not "zone" in builder:
+        printer.out("zone in cloudstack builder not found", printer.ERROR)
+        return
+    if not "description" in builder:
+        printer.out("description in cloudstack builder not found", printer.ERROR)
+        return
+
+    if "publicImage" in builder:
+        pimage.publicImage = True if (builder["publicImage"] == "true") else False
+    if "featured" in builder:
+        pimage.featuredEnabled = True if (builder["featured"] == "true") else False
+
+    pimage.displayName = builder["imageName"]
+    pimage.zoneName = builder["zone"]
+    pimage.description = builder["description"]
+    return pimage
+
+def publish_cloudstackqcow2(builder):
+    return publish_cloudstack(builder)
+
+def publish_cloudstackvhd(builder):
+    return publish_cloudstack(builder)
+
+def publish_cloudstackova(builder):
+    return publish_cloudstack(builder)
+
+
+def publish_susecloud(pimage, builder):
+    # doing field verification
+    if not "imageName" in builder:
+        printer.out("imageName in susecloud builder not found", printer.ERROR)
+        return
+    if not "tenant" in builder:
+        printer.out("tenant in susecloud builder not found", printer.ERROR)
+        return
+    if "description" in builder:
+        pimage.credAccount.description = builder["description"]
+
+    pimage.credAccount.displayName = builder["imageName"]
+    pimage.credAccount.tenantName = builder["tenant"]
+    if "publicImage" in builder:
+        pimage.credAccount.publicImage = True if (builder["publicImage"] == "true") else False
+    # if "paraVirtualMode" in builder:
```

Comment:
As you refactor this code to a new file, now you will appear responsible of the commented code ^^.